File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pseudoPalindromicPaths (self, root: Optional[TreeNode]) -> int:
        def helper(node, count=0):
            if not node:
                return 0
            count ^= 1 << node.val
            if node.left == node.right and count & (count-1) == 0:
                return 1
            return helper(node.left, count) + helper(node.right, count)
          
          
        return helper(root)

File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var pseudoPalindromicPaths = function (root) {
  return dfs(root);
};

const dfs = (node, count = 0) => {
  if (!node) return 0;

  count ^= 1 << node.val;
  if (!node.left && !node.right) return count & (count - 1) ? 0 : 1;

  return dfs(node.left, count) + dfs(node.right, count);
};
